php/media/class-global-transformations.php

```diff
@@ -272,8 +273,9 @@ public function get_taxonomy_transformations( $type ) {
 	 * @return bool
 	 */
 	public function is_taxonomy_overwrite() {
-		$apply_type = false;
-		if ( in_the_loop() ) {
+		$apply_type    = false;
+		$maybe_post_id = apply_filters( 'cloudinary_post_id_taxonomy', null );
+		if ( in_the_loop() || $maybe_post_id ) {
 			$post       = get_post();
```

Comment:
@pereirinha - I don't like the repetition here.
Perhaps something like this:
```
		$post = $this->get_current_post();
		if( $post ) { ....       
```
and place the logic in `get_current_post()`